python/GangaLHCb/Lib/Applications/GaudiExecUtils.py

```diff
@@ -87,41 +87,56 @@ def getGaudiExecInputData(optsfiles, app):
 
 exportToGPI('getGaudiExecInputData', getGaudiExecInputData, 'Functions')
 
-def prepare_cmake_app(myApp, myVer, myPath='$HOME/cmtuser', myGetpack=None):
+def prepare_cmake_app(myApp, myVer, myPath='$HOME/cmtuser', myUse=None, myFolder=None):
     """
         Short helper function for setting up minimal application environments on disk for job submission
         Args:
             myApp (str): This is the name of the app to pass to lb-dev
             myVer (str): This is the version of 'myApp' to pass to lb-dev
             myPath (str): This is where lb-dev will be run
-            myGepPack (str): This is a getpack which will be run once the lb-dev has executed
+            myUse (str): This is a git lb-use which will be run once the lb-dev has executed
+            myFolder (str): This is a git lb-checkout after the lb-use. Assumes the master branch of the use project.
     """
     full_path = expandfilename(myPath, True)
     if not path.exists(full_path):
         makedirs(full_path)
     if not path.exists(full_path + '/' + myApp + 'Dev_' +myVer):
-        _exec_cmd('lb-dev %s %s' % (myApp, myVer), full_path)
+        devStat, devOut, devErr = _exec_cmd('lb-dev %s %s' % (myApp, myVer), full_path)
+        logger.info("Running lb-dev %s %s" % (myApp, myVer))
+        if devStat != 0:
+            logger.error("lb-dev %s %s failed!" % (myApp, myVer))
+            raise GangaException(devErr)
```

Comment:
The only relevant one would be `ApplicationPrepareError` which I don't think makes much sense for a stand-alone function.